### ISSUE TITLE
Some fundamental questions about CausalML

### ISSUE THREAD

**Issue by Jami1141** [uber/causalml#177]
I have some questions about your package.

I plan to use your code to calculate CATE for each individual but when I have multiple categorical Treatments (0,1,2).
When I read your documentations. I got confused how to use it and which method to use.
Before using your package I was using CAusallift package to calculate CATE for each individual. But Causallift uses binary treatment and it does not support multi Treatment.
In your examples I don't find where to use Multi treatment and also I don't see an example for classification problem. Like XGBoost classifier for multi treatment and binary output. 

It will be great If you explain here how should I use your package step by step.

Thanks

**Comment by yungmsh** [comment]
Hi @Jami1141 thanks for your question. Our initial implementation of the multiple treatment case actually implements binary treatment analysis under the hood, looped through every treatment group. The API allows you to provide a vector of treatment assignments, and the name of the control group. For example, if you have a vector of treatment assignments `treatment` that looks something like this: `['treatment_A','treatment_A','treatment_B','control','treatment_B',...]`, and your control group is named "control", then you can call the learner like this:
```
learner_s = BaseSRegressor(XGBRegressor(), control_name='control')
ate_s = learner_s.estimate_ate(X=X, treatment=treatment, y=y)  # ATE
cate_s = learner_s.fit_predict(X=X, treatment=treatment, y=y)  # CATE
```
In the [example notebook](https://github.com/uber/causalml/blob/master/examples/meta_learners_with_synthetic_data_multiple_treatment.ipynb) for the multiple treatment case, you have to scroll about half-way down the notebook to actually get to the 3-treatment examples, under the "Multiple Treatment Case" section. Minor note that the actual synthetic data generation method for the multiple treatment case is a little hacky and should be taken with a grain of salt.

**Comment by Jami1141** [comment]
Many thanks for your reply. My output is binary, therefore my problem is classification and not regression. Then how should I implement it instead of learner_s = BaseSRegressor(XGBRegressor(), control_name='control')?
I want to use XGBoost classifier and also using metrics like roc_auc, which is used for case of classification. How can I do t?
The next question is that suppose I obtain CATE for each individual and model is trained. Now If I have new data to predict CATE what should I do? In this case, I don't have Treatment and output.

How following step should be done?
cate_s = learner_s.fit_predict(X=X, treatment=treatment, y=y)  # CATE

**Comment by yungmsh** [comment]
Assuming your `y` is a vector of 1s and 0s, try this:
```
learner = BaseSClassifier(XGBClassifier(), control_name='control')
learner.fit(X=X, treatment=treatment, y=y)
cate = learner.predict(X=X_new)
```
where X_new is your new data you want to predict CATE for.

**Comment by Jami1141** [comment]
many thanks for your reply. Then does it work the same way for other learners?
Should I try for all learners (R,X,T,..) to compare my models or depending on my problem I should choose one?
How can we evaluate cate that we obtain is good? I mean suppose we find cate for our trained data, then how we can show that our cate is good enough to use the model for later prediction?

**Comment by yungmsh** [comment]
Yeah, a typical procedure would be to run all the the learners against each other. For more on evaluation/validation, check out "Part B: Validating Meta-Learner Accuracy" in this [notebook](https://github.com/uber/causalml/blob/master/examples/meta_learners_with_synthetic_data.ipynb).

**Comment by Jami1141** [comment]
Thanks for your reply. 
I have following questions as well: 
How can we evaluate cate that we obtain is good? I mean suppose we find cate for our trained data, then how we can show that our cate is good enough to use the model for later prediction?

My next question is that, is there a way to show which treatment is the best?

**Comment by Jami1141** [comment]
Another question is that: When I use following code:

X = data3.drop(['Treatment','Outcome'],axis=1)
y = data3['Outcome']
treatment = data3['Treatment'] 
learner = BaseSClassifier(XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, gamma=0,
              learning_rate=0.1, max_delta_step=0, max_depth=3,
              min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,
              nthread=None, objective='binary:logistic', random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=32.0,
              seed=None, silent=None, subsample=1, verbosity=1), control_name='control')
learner.fit(X=X, treatment=treatment, y=y)
cate = learner.predict(X=X, treatment=treatment, y=y)
print(cate)

I get following results for cate:
INFO:causalml:Error metrics for group treatment_a
INFO:causalml:     AUC   (Control):     0.9984
INFO:causalml:     AUC (Treatment):     0.9983
INFO:causalml:Log Loss   (Control):     0.0676
INFO:causalml:Log Loss (Treatment):     0.0733
INFO:causalml:Error metrics for group treatment_b
INFO:causalml:     AUC   (Control):     0.9981
INFO:causalml:     AUC (Treatment):     0.9956
INFO:causalml:Log Loss   (Control):     0.0783
INFO:causalml:Log Loss (Treatment):     0.1100


[[1.68725383e-04 1.92721037e-03]
 [0.00000000e+00 3.08476120e-01]
 [0.00000000e+00 3.09158838e-03]
 ...
 [0.00000000e+00 2.26014969e-03]
 [0.00000000e+00 1.32014882e-03]
 [0.00000000e+00 1.01804105e-03]]

Why cate is two dimensional? each belongs to one individual in df dataframe?

**Comment by yungmsh** [comment]
> How results of above are obtained? do you separately train control, treatment and treatment b sample?

We separately run treatment vs control, and then treatment b vs control.

> Why cate is two dimensional? each belongs to one individual in df dataframe?

Your `cate.shape[1]` will be the number of treatment groups you have. You may access `learner._classes` to see what treatment corresponds to which index in cate.
